larray/inout/xw_reporting.py

```diff
@@ -0,0 +1,709 @@
+
+        >>> first_sheet = report.new_sheet('First_sheet')
+        >>> first_sheet.add_title('Title banner with default width, height and fontsize')
+        >>> first_sheet.add_title('Larger title banner', width=1200, height=100)
+        >>> first_sheet.add_title('Bigger fontsize', fontsize=13)
+
+        >>> # do not forget to call 'to_excel' to create the report file
+        >>> report.to_excel('Report.xlsx')
+        """
+        pass
+
+    def add_graph(self, title, data, template=None, width=None, height=None):
+        r"""
+        Add a graph item to the current sheet.
+        Note that the current method only add a new item to the list of items to be generated.
+        The report Excel file is generated only when the :py:obj:`~ExcelReport.to_excel` is called.
+
+        Parameters
+        ----------
+        title : string
+            title of the graph.
+        data : 1D or 2D array-like
+            1D or 2D array representing the data associated with the graph.
+            The first row represents the abscissa labels.
+            Each additional row represents a new series and must start with the name of the current series.
+        template : str, optional
+            name of the template to be used to generate the graph.
+            The full path to the template file must be provided if no template directory has not been set
+            or if the template file belongs to another directory.
+            Defaults to the defined template (see :py:obj:`~ExcelReport.set_graph_template`).
+        width : int, optional
+            width of the title item. The current default value is used if None
+            (see :py:obj:`~ExcelReport.set_item_default_size`). Defaults to None.
+        height : int, optional
+            height of the title item. The current default value is used if None
+            (see :py:obj:`~ExcelReport.set_item_default_size`). Defaults to None.
+
+        Examples
+        --------
+        >>> demo = load_example_data('demo')
+        >>> templates_dir = get_example_templates_dir()
+        >>> report = ExcelReport(templates_dir)
+
+        >>> sheet_be = report.new_sheet('Belgium')
+
+        Using default 'template', 'width' and 'height'
+
+        >>> sheet_be.add_graph('Population', demo.pop['Belgium'], template='Line')
+
+        Specifying the 'width' and 'height' values
+
+        >>> sheet_be.add_graph('Births', demo.births['Belgium'], template='Line', width=450, height=250)
+
+        Setting a default template
+
+        >>> sheet_be.template = 'Line_Marker'
+        >>> sheet_be.add_graph('Deaths', demo.deaths['Belgium'])
+
+        Dumping the report Excel file
+
+        >>> # do not forget to call 'to_excel' to create the report file
+        >>> report.to_excel('Demography_Report.xlsx')
+        """
+        pass
+
+    def newline(self):
+        r"""
+        Force a new row of graphs.
+        """
+        pass
+
+
+class AbstractExcelReport(AbstractReportClass):
```

Comment:
is this intermediate Abstract class actually needed?

Reply:
I use this trick to be able to generate the documentation on Readthedocs which runs on Linux. I prefer to do it this way rather than what we did in the xw_excel.py module.